### ISSUE TITLE
Github Actions: Optional target tests failing

### ISSUE THREAD

**Issue by slawr** [GENIVI/vss-tools#106]
The optional target tests in the Github Actions appear to be failing.

For example:
https://github.com/GENIVI/vss-tools/pull/104/checks#step:7:33

Raising it here as I could not see an open issue for this. Have not investigated further at the time of writing.

**Comment by erikbosch** [comment]
Yes, they do. It is well known and partially intentional. The optional tests (https://github.com/GENIVI/vehicle_signal_specification/blob/master/Makefile#L20) were added to make sure that all tools actually are run as part of CI. But we have quite some tools that are broken or not maintained, and some other that are in beta-stage, and we do not want to enforce that they are up to date with vss and common parts of vss-tools, i.e. it shall be possible to add an improvement to vss or vss-tools without the need to make sure that no regressions occur for contrib/optional tools. I tried to explain the difference between "contrib" and "community supported" in the readme of https://github.com/GENIVI/vss-tools/ recently.

I checked a build log of a recent build.
vspec2c and vspec2ocf are already listed as obsolete and not functional in https://github.com/GENIVI/vss-tools/, so it is expected that they will fail.

The other should to my knowledge be functional as of now, but I noticed in e.g. https://github.com/GENIVI/vss-tools/pull/103/checks?check_run_id=3688390641 that vspec2ttl complains on rdflib missing. @danielwilms - do we need to add rdflib to requirements.txt?

**Comment by danielwilms** [comment]
> The other should to my knowledge be functional as of now, but I noticed in e.g. https://github.com/GENIVI/vss-tools/pull/103/checks?check_run_id=3688390641 that vspec2ttl complains on rdflib missing. @danielwilms - do we need to add rdflib to requirements.txt?

yes, indeed. I'll create a PR.

**Comment by slawr** [comment]
> Yes, they do. It is well known and partially intentional. The optional tests (https://github.com/GENIVI/vehicle_signal_specification/blob/master/Makefile#L20) were added to make sure that all tools actually are run as part of CI. But we have quite some tools that are broken or not maintained, and some other that are in beta-stage, and we do not want to enforce that they are up to date with vss and common parts of vss-tools, i.e. it shall be possible to add an improvement to vss or vss-tools without the need to make sure that no regressions occur for contrib/optional tools. I tried to explain the difference between "contrib" and "community supported" in the readme of https://github.com/GENIVI/vss-tools/ recently.

Sorry for posting and running. Been busy. 
Thank you for the explanation. I think from the current setup I inferred that some failures were expected from the title of the containing build action, that they were optional and the action would always succeed. I also found some doc (can't remember now what I saw specifically) that showed some were deprecated. So that also was a flag that set expectation.

> 
> I checked a build log of a recent build. vspec2c and vspec2ocf are already listed as obsolete and not functional in https://github.com/GENIVI/vss-tools/, so it is expected that they will fail.
> 
> The other should to my knowledge be functional as of now, but I noticed in e.g. https://github.com/GENIVI/vss-tools/pull/103/checks?check_run_id=3688390641 that vspec2ttl complains on rdflib missing. @danielwilms - do we need to add rdflib to requirements.txt?

Ultimately I think I was surprised that there were failures across many of the optional tools and at the same time none seemed to have issues against them. So it made me wonder if some, possibly all, were not known. So I decided to raise the issue in case they were not. I see you picked up on one and an issue for that raised. I don't feel well placed to judge all the results on anything but a black and white 'build issue' basis. Now you and Daniel have reviewed the list this issue seems to have served its purpose. Shall I close?
